monitor-e2e/test/common.js

```diff
@@ -0,0 +1,87 @@
+const assert = require('assert')
+const axios = require('axios')
+const { nativeToErcBridge, ercToErcBridge, ercToNativeBridge } = require('../../e2e-commons/constants.json')
+
+const types = [
+  { description: 'NATIVE TO ERC', baseUrl: nativeToErcBridge.monitor },
+  { description: 'ERC TO ERC', baseUrl: ercToErcBridge.monitor },
+  { description: 'ERC TO NATIVE', baseUrl: ercToNativeBridge.monitor }
+]
+
+types.forEach(type => {
+  describe(type.description, () => {
+    describe('balances', async () => {
+      let data
+
+      before(async () => {
+        ;({ data } = await axios.get(`${type.baseUrl}`))
+      })
+
+      describe('home', async () => {
+        it('should contain deposits:', () => assert(data.home.deposits === 0))
+        it('should contain withdrawals', () => assert(data.home.withdrawals === 0))
+      })
+
+      describe('foreign', async () => {
+        it('should contain deposits:', () => assert(data.foreign.deposits === 0))
+        it('should contain withdrawals', () => assert(data.foreign.withdrawals === 0))
+      })
+
+      describe('general', async () => {
+        it('should contain balanceDiff', () => assert(data.balanceDiff === 0))
+        it('should contain depositsDiff', () => assert(data.depositsDiff === 0))
+        it('should contain withdrawalDiff', () => assert(data.withdrawalDiff === 0))
+        it('should contain timeDiff', () => assert(data.timeDiff >= 0))
+        it('should contain lastChecked', () => assert(data.lastChecked >= 0))
+      })
+    })
+
+    describe('validators', async () => {
+      let data
+
+      before(async () => {
+        ;({ data } = await axios.get(`${type.baseUrl}/validators`))
+      })
+
+      it('home', () => assert(typeof data.home.validators === 'object'))
+      it('foreign', () => assert(typeof data.foreign.validators === 'object'))
+      it('requiredSignaturesMatch', () => assert(data.requiredSignaturesMatch))
```

Comment:
it make sense to develop a test to check that monitor handles situations when different bridge sides have different numbers for required signatures or different validators set.

Reply:
The `REQUIRED_NUMBER_OF_VALIDATORS` and `VALIDATORS` are not set separately for different sides of the bridge, when deploying.

Seems like, in order to test different values, we would have to deploy the bridge twice with different values, and then take Home bridge from one pair and take Foreign from another pair.

This seems like overhead for just this one test case. Do we still want to do it?
In the mean time, I have extended these tests a little bit in a PR into this PR: https://github.com/poanetwork/tokenbridge/pull/192

Reply:
You can change the set of validators later if you call `addValidator` on one side of the bridge.

https://github.com/poanetwork/poa-bridge-contracts/blob/048f51e0e8e7ce4213b727c0982aadab6c994db9/contracts/upgradeable_contracts/BridgeValidators.sol#L45